### DISCUSSION TITLE
Change data capture with JanusGraph

### DISCUSSION THREAD

**Discussion by satishkotha** [JanusGraph/janusgraph#2820] [Q&A]
Hi

I want to reliably transfer all mutations performed on JanusGraph into a queue. Typically, i have seen storage systems implement change data capture. Basically, checkpointing is done on Write-Ahead-Log to ensure all events are propagated to queue reliably.  Does JanusGraph have support for this? I noticed there is "Janus Graph Bus". But from quick code scan, it seems like events are published to JanusGraph Bus after transaction is committed in asynchronous way. Because of asynchronous nature, some events may not be published to queue at scale. Is my understanding correct? 

Thanks

**Comment by satishkotha** [comment]
It looks like transaction log described here can be used for CDC https://docs.janusgraph.org/advanced-topics/transaction-log/

But I also see a note that "This log is not meant for consumption by the user or external systems - use trigger logs for that". Do trigger logs also have an option to configure 'startTime' to resume from an earlier 'processed time'?

**Reply by sandeep-mishra** [reply]
I think there is no separate trigger logs. Triggers are invoked by LogProcessor so the startTime configuration on LogProcessor do apply on triggers too.
